### PR TITLE
Add cards for majors, routes for results and careers

### LINKED ISSUE #58: Results: Major Card
image

### PR DESCRIPTION
**Pull request by Raymond0T**
#### What is the purpose of your change?
-Card component for ticket #58 
Fixes #58

#### What areas of the application does this change impact? (changes API, folder structure, etc)
- Front-end and icons folder

#### What features/fixes are deferred/not handled in this PR?
-font sizing, card icon, and page layout

#### Screenshot or link to preview
-https://deploy-preview-93--pick-a-major.netlify.app/guide
image

## Checklist
- [x] Adds unit tests
- [x] Has a work item linked to the PR

### CODE REVIEW

File: tmt-app/src/Components/MajorCard.js [hunk 1]
```diff
@@ -1,6 +1,142 @@
...
+  Card,
+  CardContent,
+  CardMedia,
+  Stack,
+  Typography,
+  Container,
+} from "@mui/material";
+import { ReactComponent as EmotionalIntelligence } from "../Images/icons/emotional_intelligence.svg";
 
 function MajorCard() {
```
Comment (michachan) [inline tmt-app/src/Components/MajorCard.js:12]: should be plural- MajorCards

File: tmt-app/src/Components/MajorCard.js [hunk 2]
```diff
@@ -1,6 +1,142 @@
...
+          width: "90%",
+          margin: "3rem auto",
+          border: "1px solid black",
+        },
+      }}
+    >
+      {dummyData.map((item) => {
+        return (
+          <Card
+            key={item.id}
```
Comment (michachan) [inline tmt-app/src/Components/MajorCard.js:43]: This should be a separate component- MajorCard

**Review by michachan — APPROVED** [review]

**Review by jwu910 — CHANGES_REQUESTED** [review]
image

Can we make some quick responsive adjustments? Some of the longer titles might be unavoidable if they're still long, but let's try to use some of the responsive typography we should have access to.

Also not sure what the interactions with these are supposed to be, should they change color when I tap on them? Are these buttons/links?

File: tmt-app/src/Components/MajorCard.js (hunk 2 above)
Comment (Raymond0T) [reply tmt-app/src/Components/MajorCard.js:43]: Are you referring to the dummyData?

**Comment by Raymond0T** [comment]
> image
> 
> Can we make some quick responsive adjustments? Some of the longer titles might be unavoidable if they're still long, but let's try to use some of the responsive typography we should have access to.
> 
> Also not sure what the interactions with these are supposed to be, should they change color when I tap on them? Are these buttons/links?

Yes, I was planning to make changes to the typography in another ticket but I could do it here. The card is suppose to change color when hovering. I will check with the team for mobile and if it would be a button or a link.

File: tmt-app/src/Components/MajorCard.js (hunk 2 above)
Comment (jwu910) [reply tmt-app/src/Components/MajorCard.js:43]: I believe Michael is suggesting you should abstract this out into its own component

File: tmt-app/src/App.js [hunk 3]
```diff
@@ -13,7 +15,15 @@ const router = createBrowserRouter([
   },
   {
     path: "/guide",
-    element: <CareerGuide />,
+    element: <ResultsPage />,
+  },
+  {
+    path: "/majorpage",
```
Comment (jwu910) [inline tmt-app/src/App.js:21]: `/majorpage` seems like a pretty odd url to see, should we have something more readable such as `majors` or `major` depending on what this page will display, also is this in scope?

File: tmt-app/src/App.js [hunk 4]
```diff
@@ -13,7 +15,15 @@ const router = createBrowserRouter([
...
+    element: <MajorPage />,
+  },
+  {
+    path: "/careerpage",
```
Comment (jwu910) [inline tmt-app/src/App.js:25]: Same comment as above

File: tmt-app/src/Components/CareerPage.jsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  const { state } = useLocation();
+
+  useEffect(() => {
+    console.log("testing", state.data);
+  }, [state]);
+
+  return <div>{state.data[0].name}</div>;
+}
+
+export default CareerPage;
```
Comment (jwu910) [inline tmt-app/src/Components/CareerPage.jsx:14]: Is this a test page? Let's not commit test code

File: tmt-app/src/Components/CustomCard.jsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                  sx={{
+                    display: "block",
+                    width: "85%",
+                    whiteSpace: "nowrap",
+                    overflow: "hidden",
+                    textOverflow: "ellipsis",
+                  }}
+                >
+                  {item.name}
+                </Typography>
```
Comment (jwu910) [inline tmt-app/src/Components/CustomCard.jsx:91]: If we hide overflow with ellipsis, is the user losing access to critical information if it's cut off? Is there another way they will be able to see this?

File: tmt-app/src/Components/CustomCard.jsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+                <EmotionalIntelligence fill="#F6A794" />
+              </Container>
+              <Typography
+                variant="CustomBody"
+                sx={{
+                  display: "-webkit-box",
```
Comment (jwu910) [inline tmt-app/src/Components/CustomCard.jsx:97]: You do not need media prefixes, what was the reason for including this?

File: tmt-app/src/Components/CustomCard.jsx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+                  overflow: "hidden",
+                  textOverflow: "ellipsis",
+                  WebkitBoxOrient: "vertical",
+                  WebkitLineClamp: "2",
+                  whiteSpace: "normal",
```
Comment (jwu910) [inline tmt-app/src/Components/CustomCard.jsx:102]: Can you explain what these styles are for?

File: tmt-app/src/Components/MajorPage.jsx [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+function MajorPage() {
+  return <div>MajorPage</div>;
+}
+
+export default MajorPage;
```
Comment (jwu910) [inline tmt-app/src/Components/MajorPage.jsx:7]: Is this page in scope?

File: tmt-app/src/Components/ResultsCards.jsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+      desc: "testing",
+      icon: "emotional_intelligence",
+    },
+    {
+      id: 1,
+      name: "Economics and Business",
+      desc: "test fdasf afas fsadf dsf dsf edsf ds fsdaf sd f asfsd fea fgdsffdsf dfasdf ds fsdafsdf fgsdaf ased fasd few gasef sWEAFGASDFAS FDFSA DFA FSADF EAFSDASD FGREAASDFG EF GRasdfga gadfaefadf asf ",
+      icon: "chemistry",
+    },
+  ];
```
Comment (jwu910) [inline tmt-app/src/Components/ResultsCards.jsx:19]: Can we create a new file with mock data to export so we can clean up what we commit in these files?

Comment (jwu910) [reply tmt-app/src/Components/ResultsCards.jsx:19]: As well as use the actual data if that is available?

File: tmt-app/src/Components/ResultsCards.jsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    <Stack
+      spacing={3}
+      sx={{
+        width: "100%",
+        "&::after": {
+          content: "''",
+          width: "90%",
+          margin: "3rem auto",
+          border: "1px solid black",
+        },
```
Comment (jwu910) [inline tmt-app/src/Components/ResultsCards.jsx:30]: Can you explain what this style is for?

**Review by jwu910 — CHANGES_REQUESTED** [review]
Seeing a lot of new changes that seem to be taking this original Pr away from the original ask. Were there accidental changes made here?

File: tmt-app/src/App.js (hunk 3 above)
Comment (Raymond0T) [reply tmt-app/src/App.js:21]: Yup, I will change the url. What do you mean by in scope?

File: tmt-app/src/Components/CareerPage.jsx (hunk 5 above)
Comment (Raymond0T) [reply tmt-app/src/Components/CareerPage.jsx:14]: This is the page that is routed to after clicking on the card. I was testing to see if I was able to successfully pass props to this page.

File: tmt-app/src/Components/CustomCard.jsx (hunk 6 above)
Comment (Raymond0T) [reply tmt-app/src/Components/CustomCard.jsx:91]: I will look for other alternatives.

File: tmt-app/src/Components/CustomCard.jsx (hunk 8 above)
Comment (Raymond0T) [reply tmt-app/src/Components/CustomCard.jsx:102]: The code for this comment and the one above would only show 2 lines max regardless of how long the description is. This will give users a short description to read and also keep all cards relatively the same size (with similar amount of lines and height).

File: tmt-app/src/Components/ResultsCards.jsx (hunk 10 above)
Comment (Raymond0T) [reply tmt-app/src/Components/ResultsCards.jsx:19]: The actual data needs some changes that would be done in a later ticket, but I could create a new file for the mock data.

File: tmt-app/src/Components/ResultsCards.jsx (hunk 11 above)
Comment (Raymond0T) [reply tmt-app/src/Components/ResultsCards.jsx:30]: This is for the solid line after the section. 
image

File: tmt-app/src/Styling/CustomStyling.js [hunk 12]
```diff
@@ -56,8 +56,23 @@ export const CustomTheme = createTheme({
...
+      "@media (max-width:1050px)": {
+        fontWeight: 700,
+        fontSize: "26px",
+        lineHeight: "50px",
+      },
+      "@media (max-width:750px)": {
+        fontWeight: 700,
+        fontSize: "20px",
+        lineHeight: "48px",
+      },
```
[the remark this answers is not in the document]
Comment (Raymond0T) [reply tmt-app/src/Styling/CustomStyling.js:75]: I will read the documentation for the default media queries and make changes accordingly.

File: tmt-app/src/App.js (hunk 3 above)
Comment (jwu910) [reply tmt-app/src/App.js:21]: Was it related to this work

File: tmt-app/src/Components/CareerPage.jsx (hunk 5 above)
Comment (jwu910) [reply tmt-app/src/Components/CareerPage.jsx:14]: Ah okay, let's leave the test code out, doesn't need to be committed.

You can look into url params for this. As long as you have url update accordingly, you should be okay

File: tmt-app/src/Components/CustomCard.jsx (hunk 8 above)
Comment (jwu910) [reply tmt-app/src/Components/CustomCard.jsx:102]: Does this work on all browsers?

File: tmt-app/src/Components/ResultsCards.jsx (hunk 11 above)
Comment (jwu910) [reply tmt-app/src/Components/ResultsCards.jsx:30]: Let's use a new element like the horizontal rule to also allow the context of this section break to exist in the code.

When looking at a page, we have different conceptual "layers", display + context in this case. We're telling the user this is the end of the section, but using styles to do so. But users using screen readers would never be able to tell. So in the context layer we could use a horizontal rule to indicate the ending of a section that a screen reader would see and communicate to the user

File: tmt-app/src/Components/CustomCard.jsx (hunk 8 above)
Comment (Raymond0T) [reply tmt-app/src/Components/CustomCard.jsx:102]: Not to sure. How would I check for all browsers?

Comment (jwu910) [reply tmt-app/src/Components/CustomCard.jsx:102]: Www.caniuse.com

Comment (jwu910) [reply tmt-app/src/Components/CustomCard.jsx:102]: Testing locally with multiple browsers is also a pretty standard self check

File: tmt-app/src/Components/CustomCard.jsx [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+} from "@mui/material";
+import { ReactComponent as EmotionalIntelligence } from "../Images/icons/emotional_intelligence.svg";
+import { Link } from "react-router-dom";
+
+function CustomCard({ dummyData }) {
+  const career = "education";
+  return (
+    <Link
+      rel="noopener noreferrer"
+      to={`/career/${career}`}
```
Comment (jwu910) [inline tmt-app/src/Components/CustomCard.jsx:17]: Did this get tested yet?

**Review by jwu910 — CHANGES_REQUESTED** [review]

File: tmt-app/src/Components/CustomCard.jsx (hunk 13 above)
Comment (Raymond0T) [reply tmt-app/src/Components/CustomCard.jsx:17]: yes, I was able to get the info I wanted after

Comment (Raymond0T) [reply tmt-app/src/Components/CustomCard.jsx]: That is expected because I was passing a constant variable for testing purposes, and I was planning to update it once the data file is complete. I will make changes to retrieve from the mock data for now.

**Comment by jwu910** [comment]
@Raymond0T are you still having issues with the icons?

File: tmt-app/src/Components/MajorCard.js (hunk 2 above)
Comment (jwu910) [reply tmt-app/src/Components/MajorCard.js]: @michachan thoughts on this? Ticket for later to clean up?

File: tmt-app/src/App.js [hunk 14]
```diff
@@ -13,7 +15,15 @@ const router = createBrowserRouter([
   },
   {
     path: "/guide",
-    element: <CareerGuide />,
+    element: <ResultsPage />,
```
Comment (jwu910) [inline tmt-app/src/App.js:18]: Is this a route that makes sense for the page we're rendering?

File: tmt-app/src/Components/CustomCard.jsx [hunk 15]
```diff
@@ -0,0 +1,124 @@
...
+import { ReactComponent as EmotionalIntelligence } from "../Images/icons/emotional_intelligence.svg";
+import { Link } from "react-router-dom";
+
+function CustomCard({ dummyData }) {
+  return (
+    <Fragment>
+      {dummyData.map((item) => {
+        return (
+          <Link
+            rel="noopener noreferrer"
```
Comment (jwu910) [inline tmt-app/src/Components/CustomCard.jsx:18]: Not necessary for internal client side routing

File: tmt-app/src/Components/CustomCard.jsx [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+  Card,
+  CardContent,
+  CardMedia,
+  Typography,
+  Container,
+} from "@mui/material";
+import { ReactComponent as EmotionalIntelligence } from "../Images/icons/emotional_intelligence.svg";
+import { Link } from "react-router-dom";
+
+function CustomCard({ dummyData }) {
```
Comment (jwu910) [inline tmt-app/src/Components/CustomCard.jsx:12]: Can we rename dummy data so it makes sense for this component?

Is there also a component name that can provide more context to what this component is and what it does?

File: tmt-app/src/Components/CustomCard.jsx [hunk 17]
```diff
@@ -0,0 +1,124 @@
...
+                ":hover": {
+                  color: "black",
+                  backgroundColor: "BTMilkTea.main",
+                },
+                cursor: "pointer",
+              }}
+            >
+              <CardMedia
+                component="img"
+                image={item.icon}
```
Comment (jwu910) [inline tmt-app/src/Components/CustomCard.jsx:41]: If we use the `component` prop to assign the `img` element, does MUI preserve access to this `image` prop still?

File: tmt-app/src/Components/CustomCard.jsx [hunk 18]
```diff
@@ -0,0 +1,124 @@
...
+                      display: "block",
+                      width: "85%",
+                      overflow: "hidden",
+                      textOverflow: "ellipsis",
+                      whiteSpace: "normal",
+                    }}
+                  >
+                    {item.name}
+                  </Typography>
+                  <EmotionalIntelligence fill="#F6A794" />
```
Comment (jwu910) [inline tmt-app/src/Components/CustomCard.jsx:92]: If we're not able to resolve the icon issue now, let's create another ticket to research and address this

File: tmt-app/src/Components/MajorCard.js
Comment (jwu910) [inline tmt-app/src/Components/MajorCard.js:1]: Why was this deleted? The currently named CustomCard wasn't what the MajorCard was right? Cause there was a separate ticket for this I think

File: tmt-app/src/Components/ResultsCards.jsx [hunk 19]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { Stack } from "@mui/material";
+import CustomCard from "./CustomCard";
+import { MOCK_DATA } from "../constants/mockData";
+
+function ResultsCards() {
+  return (
+    <Stack spacing={3}>
+      <CustomCard dummyData={MOCK_DATA} />
```
Comment (jwu910) [inline tmt-app/src/Components/ResultsCards.jsx:9]: Ah looking at this component name, I would expect one element to render, but recalling the code that's in customCard, we're passing in an array of item ms we iterate over. That logic should be here, or we should reconsider some components' names

File: tmt-app/src/Styling/CustomStyling.js [hunk 20]
```diff
@@ -37,17 +37,17 @@ export const CustomTheme = createTheme({
       fontWeight: 700,
       fontSize: "64px",
       lineHeight: "96px",
-      "@media (max-width:1350px)": {
```
Comment (jwu910) [inline tmt-app/src/Styling/CustomStyling.js:40]: Research into the theme palette object would be helpful in cleaning up some of these styles later, I think we've created some custom stuff and not leveraging MUI for what it is and what it's capable of

**Review by jwu910 — CHANGES_REQUESTED** [review]

File: tmt-app/src/Components/MajorCard.js
Comment (Raymond0T) [reply tmt-app/src/Components/MajorCard.js:1]: I believe this was meant for the results page for the CustomCard.

Comment (jwu910) [reply tmt-app/src/Components/MajorCard.js:1]: Sure no prob, as long as everyone understands the naming decisions

File: tmt-app/src/Components/ResultCards.jsx [hunk 21]
```diff
@@ -0,0 +1,130 @@
...
+import { ReactComponent as EmotionalIntelligence } from "../Images/icons/emotional_intelligence.svg";
+import { Link } from "react-router-dom";
+import TopicIcon from "../constants/SpriteSheet.svg";
+
+function ResultCards({ cardData }) {
+  const IconSvg = ({ topic, size }) => (
+    <svg fill="white" width={size} height={size}>
+      <use href={`${TopicIcon}#icon-${topic}`} />
+    </svg>
+  );
```
Comment (jwu910) [inline tmt-app/src/Components/ResultCards.jsx:18]: Nice! lets make this much more generic though. We want to consider the reuse of this elsewhere in the application for ANY icons that will be added in later. 

Couple things, the file could be just `sprite.svg` or `spriteSheet.svg` is fine too. As a nit pick and potentially functional tip, pay attention to your capitalizations of your files, variables, routes, etc. In some contexts it matters how you format them. 

I think a simple rule to follow at least for react development. PascalCasing is for components, camelCasing for variables, SNAKE_CAPS for constants, kebab-case for file naming and route paths. It can vary per project and can also be preference, but i think thats a pretty safe assumption to be had when looking at other projects too. cc: @michachan for thoughts

File: tmt-app/src/Components/ResultCards.jsx [hunk 22]
```diff
@@ -0,0 +1,130 @@
+import React, { Fragment } from "react";
+import {
+  Card,
+  CardContent,
+  CardMedia,
+  Typography,
+  Container,
+} from "@mui/material";
+import { ReactComponent as EmotionalIntelligence } from "../Images/icons/emotional_intelligence.svg";
```
Comment (jwu910) [inline tmt-app/src/Components/ResultCards.jsx:9]: Can we add this to the sprite sheet since we ended up adding the work to this PR?

File: tmt-app/src/constants/SpriteSheet.svg
Comment (jwu910) [inline tmt-app/src/constants/SpriteSheet.svg:1]: This should be in the icons directory;

how are we generating this too?

**Review by jwu910 — CHANGES_REQUESTED** [review]

File: tmt-app/src/Styling/CustomStyling.js [hunk 23]
```diff
@@ -32,81 +35,99 @@ export const CustomTheme = createTheme({
...
-      fontSize: "16px",
-      lineHeight: "24px",
+      fontSize: "14px",
+      lineHeight: "20px",
+      [breakpoints.up("lg")]: {
+        fontSize: "16px",
+      },
+      [breakpoints.up("xl")]: {
+        fontSize: "16px",
+      },
```
[the remark this answers is not in the document]
Comment (Raymond0T) [reply tmt-app/src/Styling/CustomStyling.js:130]: I believe this is going to be a body variant

File: tmt-app/src/constants/SpriteSheet.svg
Comment (Raymond0T) [reply tmt-app/src/constants/SpriteSheet.svg:1]: This would be referenced in the IconSvg component

Comment (jwu910) [reply tmt-app/src/constants/SpriteSheet.svg:1]: But how will we be adding new icons? Do you plan to do it manually?

Comment (Raymond0T) [reply tmt-app/src/constants/SpriteSheet.svg:1]: Yeah I will be doing it manually for now

**Review by michachan — APPROVED** [review]

**Review by jwu910 — APPROVED** [review]
